microservices-connector/config/samples/ChatQnA/use_cases.md

```diff
@@ -0,0 +1,222 @@
+
+   ```sh
+   kubectl create deployment client-test -n chatqa --image=python:3.8.13 -- sleep infinity
+   ```
+
+5. Access the application using the above URL from the client pod
+
+   ```sh
+   export CLIENT_POD=$(kubectl get pod -n chatqa -l app=client-test -o jsonpath={.items..metadata.name})
+   export accessUrl=$(kubectl get gmc -n chatqa -o jsonpath="{.items[?(@.metadata.name=='chatqa')].status.accessUrl}")
+   kubectl exec "$CLIENT_POD" -n chatqa -- curl -s --no-buffer $accessUrl  -X POST  -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
+   ```
+
+6. Perhaps you want to try another LLM model? Just modify the application custom resource to use another LLM model
+
+   Should you, for instance, want to change the LLM model you are using in the ChatQnA pipeline, just edit the custom resource file.
+   For example, to use Llama-2-7b-chat-hf make the following edit:
+
+   ```yaml
+   - name: Tgi
+     internalService:
+       serviceName: tgi-service-m
+       config:
+         LLM_MODEL_ID: Llama-2-7b-chat-hf
+   ```
+
+7. Apply the change
+
+   ```
+   kubectl apply -f $(pwd)/chatQnA_xeon.yaml
+   ```
+
+8. Check that the tgi-svc-deployment has been changed to use the new LLM Model
+
+   ```sh
+   kubectl get deployment tgi-service-m-deployment -n chatqa -o jsonpath="{.spec.template.spec.containers[*].env[?(@.name=='LLM_MODEL_ID')].value}"
+   ```
+
+9. Access the updated pipeline using the same URL from above using the client pod
+
+   ```sh
+   kubectl exec "$CLIENT_POD" -n chatqa -- curl -s --no-buffer $accessUrl -X POST -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
+   ```
+
+> [NOTE]
+
+You can remove your ChatQnA pipeline by executing standard Kubernetes kubectl commands to remove a custom resource. Verify it was removed by executing kubectl get pods in the chatqa namespace.
+
+### ChatQnA with data preparation
+
+This involves deploying the ChatQnA custom resource. You can use `chatQnA_dataprep_xeon.yaml` or if you have a Gaudi cluster, you could use `chatQnA_dataprep_gaudi.yaml`.
+
+1. Create namespace and deploy application
+
+   ```sh
+   kubectl create ns chatqa
+   kubectl apply -f $(pwd)/chatQnA_dataprep_xeon.yaml
+   ```
+
+2. GMC will reconcile the ChatQnA custom resource and get all related components/services ready. Check if the service up.
+
+   ```sh
+   kubectl get service -n chatqa
+   ```
+
+3. Retrieve the application access URL
+
+   ```sh
+   kubectl get gmconnectors.gmc.opea.io -n chatqa
+   NAME     URL                                                      READY     AGE
+   chatqa   http://router-service.chatqa.svc.cluster.local:8080      10/0/10    3m
+   ```
+
+> [NOTE]
+
+Comparing with `General ChatQnA with preset RAG data`, there should be `9` microservices, the extra one is the microservice of `dataprep`.
```

Comment:
Here should be 10 microservices? The above output is 10/0/10 but here mentioned 9, it's a little confusing if that's not wrong.

Reply:
Nice catch! Done!